packages/ui/src/CodeEditor.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+  CodeMirror based Code Editor component.
+
+  * For readonly/code-viewing purposes use `Code`.
+
+  ## CM5 vs CM6
+
+  We are not using the old CM5; we're basing our component on CM6 which is the next gen. version of the hugely
+  popular CM5. CM6 is still years old and solid despite being a young project.
+
+  See https://codemirror.net/6/ for more.
+
+  ## About language support
+
+  In CM6 all of the languages are defined in small, self defined packages.
+  ie. import { javascript } from '@codemirror/lang-javascript'
```

Comment:
I would suggest we use the modern packages as well to propagate the best practices. Cloud also cares about the bundle size as well.

Reply:
hmm that makes it harder to use (ok not hard but unpleasant)... you have to add the language dependencies one by one to your project then. do we want that?

Reply:
Absolutely! This will enable tree-shaking on Cloud

Reply:
hmm i see no way how tree-shaking would be ineffective with the current code (it's ONE package yes; but it's MANY language modules, you import the ones you need - so bundle size would not be that much different than our current code.)

i'm still reluctant to revert using the new packages (i did in the beginning; did not like it at all) imho we should roll with this.

Reply:
Do I understand it correctly, on the client we can do both: import legacy as well as modern language packages, (i.e. '@codemirror/legacy-modes/mode/javascript' or '@codemirror/lang-javascript')?
If so then the client will care about how big the language import is, since here we only use it in stories so the language won't be a part of the library bundle. The only drawback is one more dependency but it's not that bad.
If so then please update the comments, because when I read them I'm under impression that only the legacy languages are allowed...

Reply:
@fantkolja @aigoncharov i think my documentation confused you guys; i've touched it a bit - not mentioning "the new language modes" at all (we have no reason to use them; legacy pack gives us all we need and more). 

@fantkolja yes you're right. you can use both legacy and new language modes*. legacy language pack is already a dependency of core-ui so you WON'T NEED to add it to your dependencies manually and your bundle will contain ONLY the languages you decide to use  - that's the beauty of it. 

* you are allowed to use new language modes but then you will need to add them to your dependencies one by one. and then there're not that many of them implemented at the moment. so why bother? (you might be triggered by the word "legacy"; don't worry about that they're not going away soon.)

Reply:
@utkukaratas do you mean that the legacy languages package is three-shakable?

Reply:
@aigoncharov yes, those languages are defined in small self contained modules each.